Improved documentation by adding code comments and ensuring consistency (#1342)
collective/icalendar: docs/how-to/usage.rst

Reviewer: Whoops! Looks like we forgot to add a critical step to get the rendered view.

```suggestion
    >>> print(cal.to_ical().decode())
    BEGIN:VCALENDAR
```

Components
----------

Components are case-insensitive Python dictionaries.

> The ``icalendar.cal.Calendar`` object is a component.

Reviewer: Let's use Python object linking, where `~` displays only the last part of the full path.

```suggestion
The :class:`~icalendar.cal.Calendar` object is a component.
```

The following example shows how to set two properties for it, then display them.

.. code-block:: pycon

>     >>> cal = Calendar()                                      # create a calendar object

Reviewer: Use proper case.

```suggestion
    >>> cal = Calendar()                                      # create a Calendar object
```

Subcomponents
-------------

> Subcomponents help in organizing related calendar data such as events within a calendar structure.

Reviewer: ```suggestion
Subcomponents help organize related calendar data, such as events, within a calendar structure.
```

-----------

Property values are UTF-8 encoded strings.

> For better understanding, these values can also be converted into Python objects such as datetime for easier manipulation and computation.

Reviewer: This sentence is redundant to the ensuing discussion.

```suggestion
```

.. code-block:: pycon

    >>> cal = Calendar()                             # create a calendar object

>     >>> cal.add("dtstart", datetime(2005,4,4,8,0,0)) # set start date

Reviewer: Inline comments should have a minimum of two spaces after the line of code.

```suggestion
    >>> cal = Calendar()                              # create a calendar object
    >>> cal.add("dtstart", datetime(2005,4,4,8,0,0))  # set start date
```

Reviewer: This suggestion was omitted.

Read a calendar from a local :file:`.ics` file and view its events.

> .. .. code-block:: pycon

Reviewer: Good catch, but this is the correct syntax. Please adjust all the others as well.

```suggestion
.. code-block:: pycon
```

After loading the example file, edit and save it.

> .. .. code-block:: pycon

Reviewer: This comments out the `code-block` directive, so the ensuing code block is not rendered correctly. Please change this and all other instances as shown.

```suggestion
..  code-block:: pycon
```

Reviewer: Alternatively, rebase your branch on `main` after https://github.com/collective/icalendar/pull/1412 is merged. Thank you!
